pythonforandroid/bootstraps/pygame/__init__.py

```diff
@@ -40,21 +43,14 @@ def run_distribute(self):
             info('Copying python distribution')
             hostpython = sh.Command(self.ctx.hostpython)
             # AND: This *doesn't* need to be in arm env?
-            shprint(hostpython, '-OO', '-m', 'compileall', self.ctx.get_python_install_dir())
+            shprint(hostpython, '-OO', '-m', 'compileall', self.ctx.get_python_install_dir(),
+                    _tail=10, _filterout="^Listing", _critical=True)
```

Comment:
its new args of shprint. for pretty log optputs.
_tail mean show only N last lines of stdout. stderr will out full. like
command args | tail 10
_filterout mean do not print stdout lines matching regexp. like command
args | grep -v -e "re"
also _filter available
and _critical mean stop processing immediate after error, and do not
re-raise exception.
all this changes tot effect commands execution (except critical), just
butify its printout

2015-11-27 0:40 GMT+02:00 Alexander Taylor <notifications@github.com>:

> In pythonforandroid/bootstraps/pygame/__init__.py
> <https://github.com/kivy/python-for-android/pull/480#discussion_r46009216>
> :
>
> > @@ -40,21 +43,14 @@ def run_distribute(self):
> >              info('Copying python distribution')
> >              hostpython = sh.Command(self.ctx.hostpython)
> >              # AND: This *doesn't* need to be in arm env?
> > -            shprint(hostpython, '-OO', '-m', 'compileall', self.ctx.get_python_install_dir())
> > +            shprint(hostpython, '-OO', '-m', 'compileall', self.ctx.get_python_install_dir(),
> > +                    _tail=10, _filterout="^Listing", _critical=True)
>
> What do these extra arguments do? I thought they would be sh arguments,
> but couldn't find them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kivy/python-for-android/pull/480/files#r46009216>.
>